components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -202,6 +201,95 @@ private ApiMgtDAO() {
         return versionList;
     }
 
+    public List<String> getAPIVersionsMatchingApiNameAndOrganization(String apiName, String organizationId)
+            throws APIManagementException {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        List<String> versionList = new ArrayList<String>();
+        ResultSet resultSet = null;
+        String sqlQuery = SQLConstants.GET_VERSIONS_MATCHES_API_NAME_AND_ORGANIZATION_SQL;
+
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            ps = conn.prepareStatement(sqlQuery);
+            ps.setString(1, apiName);
+            ps.setString(2, organizationId);
+            resultSet = ps.executeQuery();
+            while (resultSet.next()) {
+                versionList.add(resultSet.getString("API_VERSION"));
+            }
+        } catch (SQLException e) {
+            handleException("Failed to get API versions matches API name" + apiName, e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(ps, conn, resultSet);
+        }
```

Comment:
Shall we use same pattern as [1] in all the DB methods
1. Dealing with AutoCommit
2. Use try with statements
3. Rolling back during an issue


[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ServiceCatalogDAO.java#L126-L162

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9609/commits/006bf23b34ccb736092e59359358e91445012850